src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -105,50 +105,12 @@ internal static partial class ICodeDefinitionFactoryExtensions
                 // to initialize all fields in the struct, not just the ones we're creating.  To
                 // do that, we call the default constructor.
 
-                var numFieldsRequiringInitialization = 0;
-
-                var autoProperties = new List<IPropertySymbol>();
-                foreach (var containingTypeField in containingTypeOpt.GetMembers().OfType<IFieldSymbol>())
-                {
-                    if (!containingTypeField.IsStatic)
-                    {
-                        numFieldsRequiringInitialization++;
-
-                        if (containingTypeField.AssociatedSymbol is IPropertySymbol autoProperty)
-                        {
-                            autoProperties.Add(autoProperty);
-                        }
-                    }
-                }
-
-                if (numFieldsRequiringInitialization == 0)
-                {
-                    return false;
-                }
-
-                foreach (var initialized in parameterToExistingFieldMap.Values)
-                {
-                    if (initialized.IsImplicitlyDeclared || initialized.IsStatic)
-                    {
-                        continue;
-                    }
-
-                    switch (initialized)
-                    {
-                        case IFieldSymbol _:
-                        case IPropertySymbol property when autoProperties.Contains(property):
-                            numFieldsRequiringInitialization--;
-                            if (numFieldsRequiringInitialization == 0)
-                            {
-                                return false;
-                            }
-                            break;
-                    }
-                }
-
-                // We have fewer field assignments than actual fields.  Generate a call to the
-                // default constructor as well.
-                return true;
+                return containingTypeOpt.GetMembers()
+                    .OfType<IFieldSymbol>()
+                    .Where(field => !field.IsStatic)
+                    .Select(field => field.AssociatedSymbol ?? field)
+                    .Except(parameterToExistingFieldMap.Values)
+                    .Any();
```

Comment:
i do like when 43 lines of code becomes 5 :)

Reply:
Depends if this path is performance sensitive. If so, one problem is that `Except` allocates a `Set` [which has 2 arrays](http://source.dot.net/#System.Linq/System/Linq/Except.cs,41)

Reply:
@Therzok This is the fix side of 'generate constructor'.  It is not perf sensitive.